.github/workflows/leeyabot.yaml

```diff
@@ -0,0 +1,58 @@
+name: Leeyabot Report
+
+on:
+  workflow_dispatch:  # Allows manual triggering from GitHub UI
+  pull_request:
+    paths:
+      - "scripts/leeyabot.Rmd"  # Runs when report.Rmd changes in a PR
+
+jobs:
+  render-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup R
+        uses: r-lib/actions/setup-r@v2
+
+      - name: Install System Dependencies (including Pandoc)
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y pandoc
+          sudo apt-get install libcurl4-openssl-dev
+          sudo apt-get install -y libjs-mathjax
+
+      - name: Cache R Packages
+        uses: actions/cache@v3
```

Comment:
It might be simpler to use `r-lib/actions/setup-r-dependencies` instead of all this stuff below? See here for an example: https://github.com/COMPASS-DOE/sensor-data-pipeline/blob/main/.github/workflows/check-workflow.yaml